src/params.rs

```diff
@@ -213,19 +254,18 @@ impl<PRFH: Hasher, MSGH: Hasher> Params<PRFH, MSGH> {
 
     // compute_chain returns the result of c(input, random_elements) iterated total times.
     fn compute_chain(
-        &mut self,
+        &self,
         p_seed: &[u8],
         input: &[u8],
         random_elements: &[Vec<u8>],
+        mut maybe_chains: Option<&mut [Vec<u8>]>,
+        idx: usize,
         begin: u8,
         end: u8,
```

Comment:
mb we can use https://doc.rust-lang.org/std/ops/struct.Range.html instead, to save this `this function has too many arguments (8/7)`, or `range: (u8, u8)`

Reply:
updated to be `range: (u8, u8)`